delivery_purchase_label/models/res_partner.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+    purchase_delivery_carrier_id = fields.Many2one(
```

Comment:
As this is used for label and not on the real transfer, I would rename to remove confusion
```suggestion
    purchase_label_carrier_id = fields.Many2one(
```